src/Trakx.IndexManager.ApiClient/AddIndexManagerApiClientExtension.cs

```diff
@@ -23,11 +24,13 @@ public static partial class AddIndexManagerApiClientExtension
             var apiConfiguration = configuration.GetSection(nameof(IndexManagerApiConfiguration))
                 .Get<IndexManagerApiConfiguration>();
             Guard.Against.NullOrEmpty(apiConfiguration.BaseUrl, nameof(apiConfiguration.BaseUrl));
-            serviceCollection.AddSingleton<ICredentialsProvider, GoogleCredentialProvider>();
-            
+            var options = Options.Create(apiConfiguration);
```

Comment:
I think this is the less verbose way to do that:
1) make sure the BaseUrl is decorated with a [Required]
Then:
```suggestion
            services.AddOptions();
            services.Configure<IndexManagerApiConfiguration>(
                Configuration.GetSection(nameof(IndexManagerApiConfiguration)));
```